resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesDriverBuilder.scala

```diff
@@ -51,7 +51,11 @@ private[spark] class KubernetesDriverBuilder(
     provideJavaStep: (
       KubernetesConf[KubernetesDriverSpecificConf]
         => JavaDriverFeatureStep) =
-    new JavaDriverFeatureStep(_)) {
+    new JavaDriverFeatureStep(_),
```

Comment:
Indentation here is super odd, but it is also odd for all the existing code. But it's kinda hard to tell what's going on.

Seems like ` new JavaDriverFeatureStep(_)` is the body of a closure, in which case it might be clearer to either keep in the same line as the `=`, or just use braces and indent the block properly.

Reply:
Argument type is a function with the default value being assigned to the default constructor. It might be cleaner to write this as follows:

- Constructor in the class declaration has _no_ default arguments
- Provide an empty arg constructor (`def this()`) which passes along all default implementations to the full-args constructor.

In this case default arguments are hurting more than they are helping, and in the unit tests, we always supply mocks for _every_ arg, meaning we always use all defaults (real implementation) or no defaults (test implementation).